crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -481,6 +481,44 @@ def upgrade_all_compute_nodes
         true
       end
 
+      def execute_scripts_and_wait_for_finish(nodes, script, seconds)
+        nodes.each do |node|
+          ssh_status = node.ssh_cmd(script).first
+          if ssh_status != 200
+            save_error_state("Failed to connect to node #{node.name}!")
+            raise "Execution of script #{script} has failed on node #{node.name}."
```

Comment:
This actually is serious error, the node cannot be accessed. I think user should be alerted immediately.